app/controllers/likes_controller.rb

```diff
@@ -1,13 +1,15 @@
 class LikesController < ApplicationController
- 
   def create
+    @like = current_user.likes.new
     @post = Post.find(params[:id])
-
-    return if @post.liked?(current_user)
-
-    @like = Like.create(author_id: current_user.id, post_id: @post.id)
-
-    redirect_to user_post_path(current_user, @post)
+    @like.post_id = @post.id
+
+    if @like.save
+      flash[:success] = 'Like Pressed'
+      redirect_to "/users/#{@post.author.id}/posts/#{@post.id}"
+    else
+      flash[:error] = 'Error : like could not be created'
+      render :new
+    end
```

Comment:
kindly prefix `.now`  to flash[:error] on line 11, to define your flash object correctly, since the message should be rendered without redirecting to another page .

Reply:
Thank you. It is noted.
I will correct it immediately